tests/e2e/utils/helpers.go

```diff
@@ -121,12 +143,12 @@ func sanitizeHTTPURL(url string) string {
 }
 
 func extractBody(r io.ReadCloser) ([]byte, error) {
+	defer r.Close()
```

Comment:
r can be `nil` then `defer r.Close()` may cause panic. So please revert it.

Reply:
I will check for nil before doing this defer. Thanks.